src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -375,22 +376,19 @@ private ListenableFuture<Boolean> checkOauth2start() {
         return oauth2started;
     }
 
-    private static class PrecheckOauth2AsyncTask extends ProgressDialogAsyncTask<Void, Void, Void> {
```

Comment:
the dialog makes no sense anymore because all the blocking things are moved to background so it is just some flicker for a fraction of a second you see the dialog and it disappears and it is confusing there is nothing to wait for

Reply:
the idea of the dialog was to provide a way to abort the things that are running in the background, so to abort getOauth2Url(() with the "Cancel" button - and this also works until MX queries were added to getProviderFromEmail() :)

however, i agree that the dialog is disturbing when things are slow (and then probably not gmail is the thing being used) - and if things are fast, the dialog is not needed